service/tablet_allocator.cc

```diff
@@ -1084,9 +1077,29 @@ class load_balancer {
                 .avg_tablet_size = avg_tablet_size,
                 .resize_decision = tmap.resize_decision(),
                 .tablet_count = tmap.tablet_count(),
-                .shard_count = shard_count
+                .shard_count = shard_count,
+                .min_tablet_count = 1,
             };
```

Comment:
I guess this min_tablet_count of 1 is needed to satisfy existing split / merge tests.

We need to write tests to stress the new heuristic.

Reply:
tablet_count for a table must be greater than or equal to 1.